rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyBooleanReturnVisitor.java

```diff
@@ -0,0 +1,181 @@
+            List<Statement> followingStatements = followingStatements();
+            Optional<Expression> singleFollowingStatement = Optional.ofNullable(followingStatements.isEmpty() ? null : followingStatements.get(0))
+                    .flatMap(stat -> Optional.ofNullable(stat instanceof J.Return ? (J.Return) stat : null))
+                    .map(J.Return::getExpr);
+
+            if (followingStatements.isEmpty() || singleFollowingStatement.map(r -> isLiteralFalse(r) || isLiteralTrue(r)).orElse(false)) {
+                J.Return retrn = getReturnIfOnlyStatementInThen(iff).orElse(null);
+                assert retrn != null;
+
+                Expression ifCondition = i.getIfCondition().getTree();
+
+                if (isLiteralTrue(retrn.getExpr())) {
+                    if (singleFollowingStatement.map(this::isLiteralFalse).orElse(false) && i.getElsePart() == null) {
+                        doAfterVisit(new DeleteStatement<>(followingStatements().get(0)));
+                        return maybeAutoFormat(retrn, retrn.withExpr(ifCondition), p, parent);
+                    } else if (!singleFollowingStatement.isPresent() &&
+                            getReturnExprIfOnlyStatementInElseThen(i).map(this::isLiteralFalse).orElse(false)) {
+                        if (i.getElsePart() != null) {
+                            doAfterVisit(new DeleteStatement<>(i.getElsePart().getBody()));
+                        }
+                        return maybeAutoFormat(retrn, retrn.withExpr(ifCondition), p, parent);
+                    }
+                } else if (isLiteralFalse(retrn.getExpr())) {
+                    boolean returnThenPart = false;
+
+                    if (singleFollowingStatement.map(this::isLiteralTrue).orElse(false) && i.getElsePart() == null) {
+                        doAfterVisit(new DeleteStatement<>(followingStatements().get(0)));
+                        returnThenPart = true;
+                    } else if (!singleFollowingStatement.isPresent() && getReturnExprIfOnlyStatementInElseThen(i)
+                            .map(this::isLiteralTrue).orElse(false)) {
+                        if (i.getElsePart() != null) {
+                            doAfterVisit(new DeleteStatement<>(i.getElsePart().getBody()));
+                        }
+                        returnThenPart = true;
+                    }
+
+                    if (returnThenPart) {
+                        // we need to NOT the expression inside the if condition
+                        Expression maybeParenthesizedCondition = ifCondition instanceof J.Binary || ifCondition instanceof J.Ternary ?
+                                new J.Parentheses<>(
+                                        randomId(),
+                                        Space.EMPTY,
+                                        Markers.EMPTY,
+                                        new JRightPadded<>(ifCondition, Space.EMPTY, Markers.EMPTY)
+                                ) :
```

Comment:
Welcoming example of replacing these `new` direction constructions w/ java template (going to look into it tomorrow solo, but just calling it out)

Reply:
fixed, able to replace all of this with a oneliner template call